src/Dropdown/Dropdown.js

```diff
@@ -121,45 +120,42 @@ export default class Dropdown extends React.Component {
     }
   };
 
-  attachAutocloseListener() {
-    if (!this.props.autoclose) {
-      return;
-    }
-    document.addEventListener('click', this.autoclose);
-  }
-
-  detachAutocloseListener() {
-    document.removeEventListener('click', this.autoclose);
-  }
-
   autoclose = e => {
-    if (this.state.isOpen && !this.menuRef.current.contains(e.target)) {
+    if (this.state.isOpen) {
       this.hide();
     }
   };
 
+  handleBlur = () => {
+    this.timeoutId = setTimeout(this.autoclose, 500);
+  }
+
+  handleFocus = () => {
+    clearTimeout(this.timeoutId);
+  }
+
   render() {
     const { width, trigger, render, children } = this.props;
     const { isOpen } = this.state;
 
     const renderFn = render || children;
 
     return (
-      <Fragment>
-        <DropdownTrigger ref={this.triggerRef} aria-haspopup="true" aria-expanded={isOpen}>
+      <DropdownWrapper tabIndex={0} onBlur={this.handleBlur}>
```

Comment:
The reference does not add a tabIndex here, but I'm not sure how to register focus or blur events on a div otherwise.